github/Repository.py

```diff
@@ -1303,7 +1304,9 @@ def create_file(self, path, message, content,
             put_parameters["author"] = author._identity
         if committer is not github.GithubObject.NotSet:
             put_parameters["committer"] = committer._identity
-
+        
+        #for replacing one or more spaces in directory names specified in  path with  '-' 
+        path =  '/'.join([re.sub(' +','-',directory) for directory in path.split('/')[0:-1]]+[path.split('/')[-1]])
```

Comment:
- This is hard to understand. What exactly does this code do?
- It looks like you are replacing spaces in all path parts but the last one. But why not in the last part?
- Why are you replace *multiple* spaces with a *single* dash?
- Also you are splitting the whole string twice.

Reply:
Replacing spaces with dash as github does .
Github doesn't allow spaces in directory names. It automatically converts spaces into dashes.

Not converting spaces in last part as last part would be filename and spaces in filename are allowed.


Yea, splitting string twice should not he done.
My bad there.